allocations/allocation_mapper.py

```diff
@@ -11,45 +34,129 @@ def _wants_db_allocations(self, request):
             return True
         return False
     
-    def map(self, project):
+    def map(self, project, get_redis_data = True):
         if self.allocations_from_db:
+            is_dict_type = isinstance(project, dict)
             balance_service = BalanceServiceClient()
             reformated_allocations = []
-            for alloc in portal_alloc.objects.filter(project_charge_code=project.chargeCode):
-                if alloc.status == 'active':
-                    active_allocation = balance_service.call(project.chargeCode)
+            project_charge_code = project['chargeCode'] if is_dict_type else project.chargeCode
+            for alloc in portal_alloc.objects.filter(project_charge_code=project_charge_code):
+                if get_redis_data and alloc.status == 'active':
+                    active_allocation = balance_service.call(project_charge_code)
```

Comment:
We are calling the balance service again here unnecessarily; I didn't catch this on the first go. Suggestion to tidy this up, and also to add logging around weird responses from the balance service:

```suggestion
                    balance = balance_service.call(project_charge_code)
                    su_used = balance.get('used')
                    if su_used is not None:
                        alloc.su_used = float(su_used)
                    else:
                        # This is an error that we should log; the balance service may be acting up
```